### PR TITLE
[CAD-402] Add mempool info.

### PR DESCRIPTION
**Pull request by ksaric**
https://jira.iohk.io/browse/CAD-402

Adding the mempool information into the trace forge events, as agreed.
Uses https://github.com/input-output-hk/ouroboros-network/pull/1433

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs [hunk 1]
```diff
@@ -30,7 +30,7 @@ import           Ouroboros.Consensus.ChainSyncServer (TraceChainSyncServerEvent)
 import           Ouroboros.Consensus.Ledger.Abstract (AnachronyFailure,
                      ProtocolLedgerView)
 import           Ouroboros.Consensus.Mempool.API (ApplyTxErr, GenTx, GenTxId,
-                     TraceEventMempool)
+                     TraceEventMempool, MempoolSize)
```
Comment (intricate) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs]: Nitpick :smiley: 

```suggestion
                     MempoolSize, TraceEventMempool)
```

**Comment by mrBliss** [comment]
@ksaric Can you rebase on master? This code has very recently been updated.

File: ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs [hunk 2]
```diff
@@ -345,6 +344,7 @@ forkBlockProduction maxBlockSizeOverride IS{..} BlockProduction{..} =
       case mIsLeader of
         Nothing    -> return ()
         Just proof -> do
+          trace $ TraceForgeAboutToLead currentSlot
```
Comment (intricate) [inline ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Good catch.

Comment (mrBliss) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: We rely on this event in the protocol tests. This is why CI is failing for this PR. 

I do admit that its name is a bit confusing. Maybe we can rename the existing trace point to `TraceForgeStartOfSlot` and use `TraceForgeAboutToLead` for the new location. /cc: @nfrisby

Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: The event needs to happen before the thread reads the ChainDB. Specifically, in the `ThreadNet` tests, the tracer itself (!!!) actually forges and adds an EBB before returning, which must be able to affect `(extLedger, (prevPoint, prevNo))`.

(... I absolutely should have added a comment here saying as much. :()

Comment (mrBliss) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Right. Do you agree that we should change its name to be slightly less misleading? The tests should of course also be updated.

Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Sure. My thoughts on the name:

  * If we consider only its role in the tests, then something extremely concrete like `TraceForgeQueringChainDBTipBeforeForging` would be very telling.
  * If we prefer a more general constructor name, then `TraceForgeSlotOnset` (with comments at its definition and use sites about needing to precede the ChainDB query) might be more reasonable.

Comment (intricate) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: Oops, I guess I got confused by the name too.

**Comment by ksaric** [comment]
I will add that when I actually run the commit that fails here, it passes locally using `stack test --fast -j12 --nix`.

File: ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs (hunk 2 above)
Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs]: FYI, @edsko renamed this tracer in PR #1459.

**Comment by mrBliss** [comment]
@ksaric Can you rebase again? :slightly_smiling_face:  As mentioned above, Edsko has recently changed this code.

**Comment by ksaric** [comment]
Rebased. It's a beauty :expressionless:

**Comment by mrBliss** [comment]
> Rebased. It's a beauty expressionless

Ah, an empty PR, I didn't realise everything you wanted to be traced is now already traced :slightly_smiling_face:. Closing.

(If the empty PR was not your intention, reopen it)

**Pull request closed without merging**